ts_sampling.py

```diff
@@ -0,0 +1,261 @@
+        linewidth=.75,
+        color="indigo",
+        label=f"Sample mean ({samples} samples)"
+    )
+    plt.plot(
+        t,
+        ci_lower,
+        "--",
+        linewidth=1,
+        color="darkorchid",
+        label=f"Gaussian {100*(1-alpha):.2f}% confidence interval"
+    )
+    plt.plot(
+        t,
+        ci_upper,
+        "--",
+        linewidth=1,
+        color="darkorchid",
+    )
+    plt.plot(
+        t,
+        mean-2*std,
+        "--",
+        linewidth=1,
+        color="orchid",
+        label="2*sample std"
+    )
+    plt.plot(
+        t,
+        mean+2*std,
+        "--",
+        linewidth=1,
+        color="orchid",
+    )
+    plt.plot(
+        t,
+        sample_fit[max],
+        color="lightskyblue",
+        label="Biggest outlier sample"
+    )
+    
+
+    plt.grid()
+    plt.xlabel("Time [year]", fontsize=12)
+    plt.ylabel("Up [mm]", fontsize=12)
+    plt.xlim([2002, 2020])
+    plt.ylim([-20, 40])
+    plt.legend(loc="lower right")
+    print(f"{station} {N_above/len(sample_gradient):.2f}% above criterion of {criterion:.1f} mm/yr")
+
+
+def plot_intervals(station, t, u, intervals=np.arange(4,11,dtype=int), samples=100, alpha=0.95):
+    
+    plt.figure(figsize=(14,8))
+    plt.plot(
+            t,
+            u,
+            ".",
+            markersize=2,
+            linewidth=0.25,
+            color="lightsalmon",
+            label="Data points",
+        )
+    for intv in intervals:
+        u_gradient, _, _, _, _ = stats.linregress(t, u)
+        sample_fit,sample_gradient=[],[]
+        for _ in range(samples):
+            sample = sample_data(data[station],intv)
+            gradient, intercept, _, _, _ = stats.linregress(
+                sample["year"], sample["U"]
+            )
+            sample_fit.append(t*gradient + intercept)
+            sample_gradient.append(gradient)
+        
+        mean,std,ci_lower,ci_upper = base_stats(
+            sample_fit,
+            intv,
+            alpha=alpha,
+        )
+        denom = np.max(intervals)*1.1
+        cicol = (1-intv/denom,1-intv/denom,intv/denom)
+        
+        plt.plot(
+            t,
+            ci_lower,
+            "--",
+            linewidth=1,
+            color=cicol,
+            label=f"2*standard deviation at {intv} intervals"
+        )
+        plt.plot(
+            t,
+            ci_upper,
+            "--",
+            linewidth=1,
+            color=cicol,
+        )
+
+    plt.plot(
+        t,
+        mean,
+        "-",
+        linewidth=0.75,
+        color="crimson",
+        label=f"Sample mean"
+    )
+
+    plt.grid()
+    plt.title(f"{(1-alpha)*100:.2f}% confidence intervals at {intervals[0:]} intervals in sampling")
+    plt.xlabel("Time [year]", fontsize=12)
+    plt.ylabel("Up [mm]", fontsize=12)
+    plt.xlim([2002, 2020])
+    plt.ylim([-20, 40])
+    plt.legend(loc="upper left")
+    print(f"{station} done with multiple intervals")
+
+
+def load_data(filename: Path):
```

Comment:
Er det her ikke samme funktion som i `gnssts.py`? Importer den i stedet for at duplikere koden